src/slackapp/flows/createDeliveryRequest.js

```diff
@@ -0,0 +1,379 @@
+          initial_channel: primaryChannel.id,
+          placeholder: {
+            type: "plain_text",
+            text: "Select a channel",
+            emoji: true
+          }
+        },
+        label: {
+          type: "plain_text",
+          text: "Channel",
+          emoji: true
+        }
+      },
+      {
+        type: "input",
+        block_id: "draft_message",
+        element: {
+          type: "plain_text_input",
+          action_id: "draft_message",
+          initial_value: suggestedTemplate(payload, request),
+          multiline: true
+        },
+        label: {
+          type: "plain_text",
+          text: "Draft Message",
+          emoji: true
+        }
+      }
+    ]
+  };
+  return requestDraftBlocks;
+}
+
+/**
+ * Modal view for confirming a delivery request
+ */
+function makeConfirmationView(request, channelId, content) {
+  const code = request.get("Code");
+  const context = {
+    code,
+    channelId,
+    content
+  };
+  return {
+    type: "modal",
+    callback_id: sendMessage.id,
+    private_metadata: JSON.stringify(context),
+    title: {
+      type: "plain_text",
+      text: modalTitle,
+      emoji: true
+    },
+    submit: {
+      type: "plain_text",
+      text: "Send",
+      emoji: true
+    },
+    close: {
+      type: "plain_text",
+      text: "Edit",
+      emoji: true
+    },
+    blocks: [
+      {
+        type: "context",
+        elements: [
+          {
+            type: "mrkdwn",
+            text: `*Request*: ${request.get(
+              "Code"
+            )}\n*Channel*: <#${channelId}>`
+          }
+        ]
+      },
+      {
+        type: "divider"
+      },
+      {
+        type: "context",
+        elements: [
+          {
+            type: "mrkdwn",
+            text: content
+          }
+        ]
+      }
+    ]
+  };
+}
+
+/**
+ * Generates the suggested slack message given a request
+ */
+function suggestedTemplate(payload, request) {
+  const slackId = payload.user.id;
+  const streets = [
+    request.get("Cross Street #1"),
+    request.get("Cross Street #2")
+  ].join(" & ");
+
+  const name = `our neighbor at ${streets}`;
+  let quadrant = request.get("Neighborhood Area (See Map)");
+  if (quadrant.match(/^NE|SE|SW|NE$/)) {
+    quadrant += " Crown Heights";
+  }
+
+  const services = request.get("What type(s) of support are you seeking?");
+  let aidType = "for aid from";
+  let needs = services.join(", ");
+  if (isEqual(services, ["Deliver groceries or supplies to me"])) {
+    // eslint-disable-line
+    aidType = "to deliver groceries to";
+    needs = "Groceries / Shopping";
+  }
+
+  const extraFields = [
+    ["Timeline", request.get("Time Sensitivity") || "When convenient"],
+    ["Neighborhood", quadrant],
+    ["Need", needs],
+    ["Cross Streets", streets],
+    ["Language", (request.get("Languages") || []).join("/")],
+    ["Description", request.get("Intake General Notes")],
+    ["Code", request.get("Code")]
```

Comment:
Know of any clever ways to make this easier to find with regex? My delivery shortcut regex is failing right now

Reply:
Looks like you fixed it?

Reply:
Maybe `/code[^\w\d]+(?<code>[\w\d]{4})$/im`? That way you don't have to manually futz with the length of the capture?

Reply:
Sounds good. Resolved